api/pull.go

```diff
@@ -221,6 +221,10 @@ func (api *PullAPI) UninstallFilter(id rpc.ID) bool {
 	api.mux.Lock()
```

Comment:
I think that the `filterExpiryChecker` goroutine, must be also a place making lots of lock acquisitions. https://github.com/onflow/flow-evm-gateway/blob/db59557c2b710af7cd32ca7d24efc00da6112d66/api/pull.go#L382-L394. Even though we call `api.uninstallFilter(id)` on the relevant endpoints, the reason we need this goroutine is because the `api.filters` might get out of capacity (the limit is `10_000` filters). Maybe we could improve this?

Reply:
It's not a problem if there are a lot of lock acquisitions, the timeout was coming from a dead-lock not a lot of pending locks. The pending locks would resolve at some point, and since the uninstall filter is just a deletion on a map we are talking about nanoseconds here, so I wouldn't go optimizing this... if the operations are fast obtaining a lock and make other wait for it is not wrong... if operations become slow we should optimize but this I feel would be a too early optimization... also it's not that all filters are uninstalled, only expired, which could be few per each round, and each round is one every few minutes so this is really not candidate for optimization imho